src/main/python/ayab/preferences.py

```diff
@@ -151,7 +176,7 @@ def __init__(self, parent):
         # add form items
         self.__widget = {}
         for var in self.__prefs.variables.keys():
-            self.__widget[var] = self.__make_widget(var)
+            self.__widget[var] = self.__make_widget(cast(PreferencesDictKeys,var))
```

Comment:
`TypedDict.keys()` isn't Generic so even though it should know that `var` is a `PreferencesDictKeys`, it don't. So we help it out with an explicit cast.